### ISSUE TITLE
Option to use short product description instead of full description

### ISSUE THREAD

**Issue by nima-karimi** [woocommerce/google-listings-and-ads#782]
We currently use the short description of a product only if the full description is not set. An option could be provided for the user to choose which description they want to use for their Google listing.

This has been already requested by some users in the WordPress.org forums: [#](https://wordpress.org/support/topic/overall-plugin-%d1%81ritique/) , [#](https://wordpress.org/support/topic/product-description-31/) , and [#](https://wordpress.org/support/topic/overall-plugin-%d1%81ritique/#post-14553312)

**Comment by mikkamp** [comment]
Can we just add a simple tab for our plugin to WooCommerce > Settings > Integrations? Or do we need a more elegant solution for a settings page?

**Comment by findingsimple** [comment]
@j111q (and Shiva - sorry not sure what your github username is if you have one) - what are your thoughts on where settings like this would live in GLA?

Quick fix is adding a filter

**Comment by j111q** [comment]
CC @shivanarrthine

> Quick fix is adding a filter

Not sure what you mean by this - What type of filter would this be?

**Comment by mikkamp** [comment]
A filter would be a small code snippet they could put on their site to change the way the plugin works. Something like this:

```
add_filter( 'woocommerce_gla_use_product_short_description', '__return_true' );
```

Easy for us to implement, but not a friendly UI.

**Comment by j111q** [comment]
Ah I seeeee. 

To be honest, I would imagine using the short description is _always_ preferred, given how that the text displayed in Google is always truncated as far as I can see. 

Even if you click into a product's details, Google Shopping seems to hide long text under a "More" link. 

 I would wonder if there's a use case for ever choosing the standard description over the short one. 🤔 

I'm leaning towards a solution of:

- If there's a short description, use the **short** description.
- If there's **no** short description, use the **standard** description. 

Added this to the agenda to chat with the Googlers about it this week. Let me know what y'all think though!

**Comment by mikkamp** [comment]
I personally think always [using short description](https://github.com/woocommerce/google-listings-and-ads/pull/783#pullrequestreview-682820327) is the best, but interested to hear other thoughts.

**Comment by findingsimple** [comment]
When checked previously the GPF extension defaulted to full (I'm fairly sure but could be wrong), perhaps some logic there worth checking.

**Comment by mikkamp** [comment]
It does seem to default to (full preferred), although it has the possibility to select many options, and this can be set for all products at once:
![image](https://user-images.githubusercontent.com/11388669/123749412-4dbb6280-d8ad-11eb-8ea4-695b9817b856.png)

**Comment by findingsimple** [comment]
Thanks for confirming. Yeah so I think [something that mapping](https://github.com/woocommerce/google-listings-and-ads/issues/572) is where we eventually want to get to (priorities pending) but given the feedback so far we should look at a quick tweak.

I think we can flip to using the short description as default but will want to do a notice or similar to warn existing users that may be expecting full now (and provide them with a path to maintain full)

**Comment by mikkamp** [comment]
> provide them with a path to maintain full

Implementation wise, what would that look like? Give them a code snippet to revert back to full? Have a settings page where they can set it back to full?

**Comment by j111q** [comment]
> I think we can flip to using the short description as default but will want to do a notice or similar to warn existing users that may be expecting full now (and provide them with a path to maintain full)

Assuming the Google team confirms my hunch that a short description is best practice, I think we should default to `short description`, and add documentation in the release notes / knowledge base... and then just see if we get any feedback requesting for the standard description instead. If we do, we can provide a code snippet to show the full description. 

I'm not sure we _should_ support all the options that Google Product Feed has, because it seems fairly convoluted and I would genuinely wonder what's the use case of each. 🤔  I think it might be better for us to set the "best practice" approach, rather than allow user customizability.

Anyway, very open to hearing other opinions here for sure. I've also noted it down as a potential research question for our upcoming research sessions.

**Comment by findingsimple** [comment]
> Give them a code snippet to revert back to full? Have a settings page where they can set it back to full?

Yep, either of those.

> I'm not sure we should support all the options that Google Product Feed has because it seems fairly convoluted and I would genuinely wonder what's the use case of each. 🤔 I think it might be better for us to set the "best practice" approach, rather than allow user customizability.

"Best practice" is fine - I'm not pushing either way for the short or long description.

Use case would be merchants have existing attributes (or data that can be used for filling in those attributes) added using another extension or something custom. The approach in GPF basically allows them to set the source of our attributes to be something else.

I'm not sure what specifically mean by _each_ -  GPF just basically allows any of the attributes to be set to use the data from an existing field or attribute rather than filling everything from scratch (as an example) or having a single field for multiple plugins (which IMO is what we would want to do for all our extensions that use product data - i.e. Facebook, Pinterest) - rather than them all adding their own.

Again, I'm not arguing either way but "WooCommerce is the most customizable eCommerce platform for building your online business." so I don't think we should lose sight of that. Adding a little bit of flexibility if done right can be a powerful thing

**Comment by mikkamp** [comment]
Another perspective to take into account is that we allow setting of about 15 product attributes per product/variation (all saved in post meta). Allowing those attributes to come from different sources, or be the same across various extensions is better for scaling (it can get out of hand quite quickly if each solution has a different way of saving those attributes).

WooCommerce itself already has built in product attributes, but if they are setup on a site we don't allow mapping between the two (until we implement #572).

In my opinion the flexibility (and simplicity for the merchant) which is allowed in GPF is one of it's strong points when it comes to setting up the product attributes.

**Comment by j111q** [comment]
Hi team, I discussed this with the Googlers this morning. 

For now, let's implement:

1. Use full description
2. If no full description, then use short description

Googlers preferred a full description to be the default over the short description, as they would prefer more data to be sent rather than less data. 

This should help anyone whose products are being rejected for a blank full description. ([A user, Mark Ross, mentions it in his comment](https://ideas.woocommerce.com/forums/133476-woocommerce/suggestions/43607094-option-to-select-description-short-or-standard))

For users who want more granular control and who want to configure this, we will look at that feature holistically with the experience of mapping other product attributes to the product feed, for a future release. 

LMK if you have additional questions!

**Comment by mikkamp** [comment]
We already implemented "use short description if full is empty" in #783 (released in 1.1.0). The comment you linked to was from before that release, so I imagine they aren't experiencing that anymore.

So I guess this will need to be addressed once we decide how to allow more granular control.

**Comment by j111q** [comment]
Oh okay great 😁  Good that we did it that way - Googlers were not keen to go with the short description over the full one. Thanks @mikkamp!

**Comment by findingsimple** [comment]
Stick with the filter for now implemented in #879 based on "Googlers were not keen to go with the short description over the full one."
